core/src/main/java/io/hyperfoil/core/steps/ScheduleDelayStep.java

```diff
@@ -298,44 +298,23 @@ public ThinkTimeBuilder init(String param) {
       }
    }
 
-   private static class ConstantDuration implements SerializableToLongFunction<Session> {
-      private final long duration;
-
-      ConstantDuration(long duration) {
-         this.duration = duration;
-      }
+   private record ConstantDuration(long duration) implements SerializableToLongFunction<Session> {
 
       @Override
       public long applyAsLong(Session session) {
          return duration;
       }
    }
 
-   private static class RandomLinearDuration implements SerializableToLongFunction<Session> {
-      private final long min;
-      private final long max;
-
-      RandomLinearDuration(long min, long max) {
-         this.min = min;
-         this.max = max;
-      }
+   private record RandomLinearDuration(long min, long max) implements SerializableToLongFunction<Session> {
```

Comment:
We do perform reflective accesses or any other serialization we expect to break making this a record?

Reply:
revert just to be sure as not really needed here